File: Upcoder/Other/Lũy thừa/Luythuatb.cs
using System;
namespace Luythua
{
    public class Program
    {
        const int MOD = 1000000007;

        // Hàm lũy thừa nhanh trong C#
        static long PowerMod(long a, long b, int mod)
        {
            long result = 1;
            a = a % mod;
            while (b > 0)
            {
                if ((b & 1) == 1) // Nếu b lẻ
                {
                    result = (result * a) % mod;
                }
                a = (a * a) % mod; // Nhân đôi cơ số
                b >>= 1;           // Chia b cho 2
            }
            return result;
        }

        static void Main(string[]args)
        {
            string[] input = Console.ReadLine().Split();
            long a = long.Parse(input[0]);
            long b = long.Parse(input[1]);
            Console.Write(PowerMod(a, b, MOD));
        }
    }
}

File: Upcoder/Other/Lũy thừa/Luythuatb.cpp
#include <bits/stdc++.h>
using namespace std;

const int MOD = 1e9 + 7;

long long powerMod(long long a, long long b, int mod) {
    long long result = 1;
    a = a % mod;
    while (b > 0) {
        if (b % 2 == 1) { 
            result = (result * a) % mod;
        }
        a = (a * a) % mod;  
        b /= 2;             
    }
    return result;
}
int main() {
    long long a, b;
    cin >> a >> b;
    cout << powerMod(a, b, MOD);
    return 0;
}
